kratos/utilities/multi_level_data_accesors.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include "containers/multi_level_data_value_container.h"
+
+namespace Kratos
+{
+
+/**
+ * @class LayeredGaussPointDataAccesor
+ * @brief A class for accessing multi-level data in a layered Gauss point structure.
+ * This class provides functionality to access data organized in multiple layers,
+ * each containing a number of Gauss points. It includes methods to compute the
+ * index based on layer and Gauss point indices, and to retrieve the total size,
+ * number of layers, and number of Gauss points.
+ * @note This class inherits from MultiLevelDataValueAccesor.
+ * @tparam NumLayers The number of layers in the data structure.
+ * @tparam NumGaussPoints The number of Gauss points in each layer.
+ */
+class KRATOS_API(KRATOS_CORE) LayeredGaussPointDataAccesor
+ : public MultiLevelDataValueAccesor
```

Comment:
```suggestion
class KRATOS_API(KRATOS_CORE) LayeredGaussPointDataAccessor
 : public MultiLevelDataValueAccessor
```

Better fix it early, or we'll end up like with _auxiliar_.

Reply:
?